src/wasi/Wasi.h

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2023-present Samsung Electronics Co., Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Walrus.h"
+#include "runtime/Value.h"
+#include "runtime/Function.h"
+#include "runtime/ObjectType.h"
+
+namespace Walrus {
+
+class WASI {
+public:
+    WASI()
+    {
+    }
+
+    struct WasiFunc {
+        std::string name;
+        FunctionType* functionType;
+        ImportedFunction::ImportedFunctionCallback ptr;
+    };
+
+// F(function_name, arguments, RESULTS, results) -- RESULTS is a way to differentiate the two inside the definition
+#define FOR_EACH_WASI_FUNC(F) \
+    F(test, RESULT)           \
+    F(printI32, I32, RESULT)  \
+    F(writeI32, RESULT, I32)
+
+    enum WasiFuncName : size_t {
+#define DECLARE_FUNCTION(NAME, ...) NAME##FUNC,
+        FOR_EACH_WASI_FUNC(DECLARE_FUNCTION)
+#undef DECLARE_FUNCTION
+            FuncEnd,
+    };
+
+    static FunctionType* determineFuncionType(std::string str)
```

Comment:
Typo `Funcion` in this method name

Reply:
One more)
Instead of creating each FunctionType for WASI function
What about reusing `SpecTestFunctionTypes` defined in Shell.cpp?

Reply:
It can be done, I am just unsure if there are wasi functions where either the amount of arguments or results can be variable and we need to create new FunctionTypes, so just in case I created this function.

Reply:
Yes, please reuse `SpecTestFunctionTypes` as we already share a lot of it.
If we need new FunctionTypes in the future, then we can just add types on `SpecTestFunctionTypes`.

Reply:
I moved `SpecTestFunctionTypes` to a different file and also reworked the wasi functions to use those types. I am also thinking that renaming those to something like `ConstantFunctionTypes` could help to understand the code later down the line. What is your opinion on that?